### ISSUE TITLE
Generated Json class is missing toString() which results in compile error

### ISSUE THREAD

**Issue by divath** [immutables/immutables#406]
When toString() is missing from an abstract class extending other class I am getting following warning:

```
    Warning:(10, 17) java: Type inherits non-default 'toString' method but have some non-inherited attributes.
    Please override 'toString' with abstract method to have generate it.
    Otherwise override with calling super implemtation to use custom implementation
```

In result toString() is not generated in ImmutableTest.

However if I try overriding toString as suggested in the warning I am getting compile errors since nonabstract final ImmutableTest.Json is missing toString() method.

Example which generates compile error:

```
@JsonSerialize(as = ImmutableTest.class)
@JsonDeserialize(as = ImmutableTest.class)
@Value.Immutable
public abstract class Test extends Exception {
    public abstract long test();

    @Override
    public abstract String toString();
}

```

**Comment by elucash** [comment]
Thank you for reporting this!

**Comment by elucash** [comment]
released in 2.2.11 Thank you!

**Comment by divath** [comment]
Thanks, verified and it works!
